### PR TITLE
Consolidate method signature for `PlatformDispatcher.onError`

### PR DESCRIPTION
**Pull request by ueman**
I'm currently writing an integration which reports `PlatformDispatcher.onError` errors directly to Sentry, see https://github.com/getsentry/sentry-dart/pull/915.
During that process, I've noticed that the method signatures differ between web and Dart:IO.
I'm pretty sure the method signature should be the same on both platforms.

On dart:ui it's

https://github.com/flutter/engine/blob/28a5d3c05d61b25f82080260bae4b6c3b77d03b5/lib/ui/platform_dispatcher.dart#L64

and on dart:web_ui it's

https://github.com/flutter/engine/blob/28a5d3c05d61b25f82080260bae4b6c3b77d03b5/lib/web_ui/lib/platform_dispatcher.dart#L17

This PR consolidates the callback to `typedef ErrorCallback = bool Function(Object exception, StackTrace stackTrace);`.

I guess this should have a test, but I'm not sure what the right approach for that is. @dnfield do you have an idea, since you've written the original implementation?

**List which issues are fixed by this PR. You must list at least one issue.**
This helps with https://github.com/getsentry/sentry-dart/pull/915. The issue this PR fixes is not really complicated, so I hope that's enough. Otherwise, I'm happy to create an issue.

This also fixes https://github.com/flutter/flutter/issues/107035

**If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].**
No, this is not a breaking change.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by dnfield** [comment]
This was a mistake. @jonahwilliams or @yjbanov could the api conformance check catch this?

**Comment by jonahwilliams** [comment]
I'm not sure, I haven't touched it since pre-null safety. I would just update the web ui version. The dart:ui signature is what is used for analysis.

File: lib/web_ui/lib/platform_dispatcher.dart [hunk 1]
```diff
@@ -8,13 +8,13 @@ typedef VoidCallback = void Function();
 typedef FrameCallback = void Function(Duration duration);
 typedef TimingsCallback = void Function(List<FrameTiming> timings);
 typedef PointerDataPacketCallback = void Function(PointerDataPacket packet);
-typedef KeyDataCallback = bool Function(KeyData packet);
+typedef KeyDataCallback = bool Function(KeyData data);
```
Comment (ueman) [inline lib/web_ui/lib/platform_dispatcher.dart:11]: This was caught by the new typedef conformance check

**Comment by ueman** [comment]
I've added a conformance check for typedef and also for nullability. Since this is my first time working with the analyzer, I'm pretty sure that this can be improved.

**Review by dnfield — APPROVED** [review]

**Pull request merged**